crates/codegen/language/definition/src/model/utils/version_specifier.rs

```diff
@@ -20,4 +20,16 @@ impl VersionSpecifier {
             VersionSpecifier::Range { from, till } => from <= version && version < till,
         }
     }
+
+    /// Returns an iterator over the versions specified by this specifier.
+    pub fn versions(&self) -> impl Iterator<Item = &Version> {
+        match self {
+            VersionSpecifier::Never => [None, None],
+            VersionSpecifier::From { from } => [Some(from), None],
+            VersionSpecifier::Till { till } => [None, Some(till)],
+            VersionSpecifier::Range { from, till } => [Some(from), Some(till)],
+        }
+        .into_iter()
+        .flatten()
```

Comment:
I wonder what is the advantage of using slices of `Option<>` + `flatten()`, rather than using something like `vec![]` directly with the final result?

just a nit suggestion for readability. Not blocking of course.

```suggestion
    match self {
        VersionSpecifier::Never => vec![],
        VersionSpecifier::From { from } => vec![from],
        VersionSpecifier::Till { till } => vec![till],
        VersionSpecifier::Range { from, till } => vec![from, till],
    }
```

Reply:
Also, I suggest specifing in the name that these are the lower/upper bounds, not all "versions specified by this specifier". Given that `Range(1, 5).versions()` will not include `2` for example.

Reply:
Re `vec`s, I wanted to avoid wasteful multiple single-or-two-element vec heap allocations only to collected into a set immediately.

We could: 
1. pull and use `SmallVec<[_; 2]>` here or
2. just bite the bullet and use `vec!` directly,

however, since it's a very easy perf win and I plan to replace this code immediately with `Language::collect_breaking_changes` (as I mentioned in the OP), I'd prefer if we don't change it/not focus on this specific snippet in this PR.

Reply:
Sounds good. Thanks for explaining.